templates/master/00-master/_base/files/usr-local-bin-etcd-member-add-sh.yaml

```diff
@@ -0,0 +1,72 @@
+filesystem: "root"
+mode: 0755
+path: "/usr/local/bin/etcd-member-add.sh"
+contents:
+  inline: |
+    #!/usr/bin/env bash
+    
+    # example
+    # export SETUP_ETCD_ENVIRONMENT=$(oc adm release info --image-for setup-etcd-environment --registry-config=./config.json)
+    # export KUBE_CLIENT_AGENT=$(oc adm release info --image-for kube-client-agent --registry-config=./config.json)
+    # sudo -E ./etcd-member-add.sh 192.168.1.100 $etcd_name
+    
+    if [[ $EUID -ne 0 ]]; then
+      echo "This script must be run as root"
+      exit 1
+    fi
+    
+    : ${SETUP_ETCD_ENVIRONMENT:?"Need to set SETUP_ETCD_ENVIRONMENT"}
+    : ${KUBE_CLIENT_AGENT:?"Need to set KUBE_CLIENT_AGENT"}
+    
+    usage () {
+        echo 'Active server IP address and etcd name required: ./etcd-member-add.sh 192.168.1.100 $etcd_name'
+        exit 1
+    }
+    
+    if [ "$1" == "" ] || [ "$2" == "" ]; then
+        usage
+    fi
+    
+    RECOVERY_SERVER_IP=$1
+    ETCD_NAME=$2
+    
+    ASSET_DIR=./assets
+    ASSET_DIR_TMP="$ASSET_DIR/tmp"
+    CONFIG_FILE_DIR=/etc/kubernetes
+    MANIFEST_DIR="${CONFIG_FILE_DIR}/manifests"
+    RUN_ENV=/run/etcd/environment
+    
+    MANIFEST_STOPPED_DIR="$ASSET_DIR/manifests-stopped"
+    ETCD_MANIFEST="${MANIFEST_DIR}/etcd-member.yaml"
+    ETCD_CONFIG=/etc/etcd/etcd.conf
+    ETCDCTL=$ASSET_DIR/bin/etcdctl
+    ETCD_VERSION=v3.3.10
+    ETCD_DATA_DIR=/var/lib/etcd
+    ETCD_STATIC_RESOURCES="${CONFIG_FILE_DIR}/static-pod-resources/etcd-member"
+    
+    SHARED=/usr/local/share/openshift-recovery
+    
+    source "/usr/local/bin/openshift-recovery-tools"
+    
+    function run {
+      init
+      dl_etcdctl
+      backup_manifest
+      DISCOVERY_DOMAIN=$(grep -oP '(?<=discovery-srv=).*[^"]' $ASSET_DIR/backup/etcd-member.yaml )
+      if [ -z "$DISCOVERY_DOMAIN" ]; then
+        echo "Discovery domain can not be extracted from $ASSET_DIR/backup/etcd-member.yaml"
+        exit 1
+      fi
+      validate_environment
+      source  /run/etcd/environment
+      backup_etcd_conf
+      backup_etcd_client_certs
```

Comment:
We probably don't need to back certs at all, as we are not touching. Perhaps a test to validate the certs would, instead, be a good idea.

Reply:
I was thinking of a test to validate the certs at the very beginning, and if the certs are not valid, exit with an appropriate error message. WDYT?

Reply:
The assumption for the script is the certs are valid. If we can't assume that should they just use the current restore that rolls new certs?